contracts/mixnet/src/transactions.rs

```diff
@@ -29,42 +31,25 @@ pub(crate) const MINIMUM_BLOCK_AGE_FOR_REWARDING: u64 = 17280;
 // 5. The node rebonds with the same identity
 pub fn delegations<T: DeserializeOwned + Serialize>(
     delegations_bucket: ReadonlyBucket<T>,
-) -> StdResult<Vec<(Vec<u8>, T)>> {
+) -> StdResult<Vec<UnpackedDelegation<T>>> {
     // I think it's incredibly unlikely to ever read more than that
     // but in case we do, we should guard ourselves against possible
     // out of memory errors (wasm contracts can only allocate at most 2MB
     // of RAM, so we don't want to box the entire iterator)
     let mut delegations = Vec::new();
-    // let mut total_delegation = Coin::new(0, DENOM);
-    let mut start = None;
+    let mut start_after = None;
```

Comment:
I haven't been closely watching changes to this file, but big red flag I can see right now is that you're putting everything into the `delegations` vector. We don't want to do this - if some node has a lot of delegations this will cause an OOM. we need to somehow stagger it or something...

Reply:
I’ve put that one in in order to be able to overwrite delegations of the old type. Given the OOM concern is there a better place to have something like this for migration purposes?

Reply:
As per the offline discussion, I will look into having it as an iterator, so that it doesn't consume memory for storing all the delegations.